Sources/NIOPosix/PendingDatagramWritesManager.swift

```diff
@@ -77,6 +76,13 @@ private func doPendingDatagramWriteVectorOperation(pending: PendingDatagramWrite
     var c = 0
     var toWrite: Int = 0
 
+    let buffer = bufferPool.get()
+    defer { bufferPool.put(buffer) }
+
+    let count = (buffer.count / (MemoryLayout<IOVector>.stride + MemoryLayout<Unmanaged<AnyObject>>.stride))
+    let iovecs = (buffer.baseAddress!).bindMemory(to: IOVector.self, capacity: count)
+    let storageRefs = (buffer.baseAddress! + (count * MemoryLayout<IOVector>.stride)).bindMemory(to: Unmanaged<AnyObject>.self, capacity: count)
```

Comment:
This code is very precise. Can we factor it out into its own method to which we can add assertions on appropriate length?

Reply:
So what you're doing here is allocating a single buffer that you then slice into the relevant buffers we want. That's fine, but I'd like to encapsulate all of that dance into a struct that we can use to hide this mess of memory binding. It will also let us add a bunch of pseudo-bounds-checks to our construction code, so we can confirm that we're not actually binding or using memory that is out of bounds of the original allocation. Does that make sense?

Reply:
well,
I would look at that with a very simple metric: number of code lines and number of extra entities (we will need to introduce an additional structure). Here we have just 3 lines of code, adding a struct with that functionality will give more than 3 lines... But if you think it is better in that particular case - let's do it.

Reply:
Generally I don't think lines of code added is the particular metric I'm most concerned about. What I want is to build abstractions that discourage misuse. A related opportunity is to actively reference count the pointers, but I think this is a one-step-at-a-time kind of deal, so I'm interested for now in giving us a sensible abstraction.